### PR TITLE
Add task source for media element

### LINKED ISSUE #21612: Implement media element event task source
To be used [here](https://github.com/servo/servo/blob/master/components/script/dom/htmlmediaelement.rs#L53)

https://html.spec.whatwg.org/multipage/media.html#media-element-event-task-source

### PR DESCRIPTION
**Pull request by chansuke**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #21612 .
- [x] These changes do not require tests because the existing test should cover this code.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/21879)

### CODE REVIEW

File: components/script/dom/htmlmediaelement.rs [hunk 1]
```diff
@@ -619,9 +615,7 @@ impl HTMLMediaElement {
         let this = Trusted::new(self);
         let generation_id = self.generation_id.get();
         self.take_pending_play_promises(Err(Error::NotSupported));
-        // FIXME(nox): Why are errors silenced here?
```
Comment (ferjm) [inline components/script/dom/htmlmediaelement.rs]: This FIXME is still valid :) Do not remove it, please.

**Review by ferjm — CHANGES_REQUESTED** [review]
Thanks for working on this @chansuke :)

The task source related changes looks good to me. However, it would be great if you could remove the second commit and rebase on top of master instead of creating a merge commit.

Thanks!

**Comment by chansuke** [comment]
@ferjm Ok.Thanks for your review.

File: components/script/task_source/media_element.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+use script_thread::MainThreadScriptMsg;
+use servo_atoms::Atom;
+use std::fmt;
+use std::result::Result;
+use std::sync::mpsc::Sender;
+use task::{TaskCanceller, TaskOnce};
+use task_source::{TaskSource, TaskSourceName};
+
+#[derive(Clone, JSTraceable)]
+pub struct MediaElementTaskSource(pub Sender<MainThreadScriptMsg>, pub PipelineId);
```
Comment (ferjm) [inline components/script/task_source/media_element.rs]: This `Sender` needs to be a `servo_channel::Sender` instead of a `std::sync::mpsc::Sender`

**Review by ferjm — CHANGES_REQUESTED** [review]
Looks better now :) Thanks for the rebase.

I think there are only two things left to fix here:

- The `servo_channel::Sender` related build error.
- These tidy issues https://travis-ci.org/servo/servo/jobs/439453305#L447

**Comment by ferjm** [comment]
One last change ;)

https://travis-ci.org/servo/servo/jobs/439558351#L447

Tip: you can check that tidy is happy with the patch by running `./mach test-tidy` locally.

**Comment by CYBAI** [comment]
@bors-servo retry

-  #21205

**Pull request merged**